### PR TITLE
Create relationships between User/Wishlist and Wishlist/Item models

### LINKED ISSUE #9: [Dependent on Issue #6, #7] Add relationships between User/Wishlist and Wishlist/Item
User has_many wishlists.
Wishlist belongs_to user.
Item belongs_to wishlist.
Wishlist has_many items.

### LINKED ISSUE #36: [Dependent on Issues #7, #6] Create relation between User and Wishlist models
Add reference column user_id to Wishlist model.

### PR DESCRIPTION
**Pull request by agalkiewicz**
Closes #9.
Closes #36.

### CODE REVIEW

**Comment by nodzy** [comment]
I think there's mistake in a name of an issue - it's relationship between User and Wishlist, not Item and Wishlist :)
In my opinion everything's fine, you'll just have to update your local master and re-generate schema, as I merged generate-item brange few minutes ago.

**Comment by student-tomasz** [comment]
Why don't we add the one-to-many association between Wishlists and Items here as well?

**Comment by agalkiewicz** [comment]
I will change the commit name and add another associaton, as You said.

**Comment by student-tomasz** [comment]
Just a question. I don't see an Issue for that, and this association is missing?

**Comment by agalkiewicz** [comment]
It looks like it is missing, so I'm doing it right now.

**Comment by agalkiewicz** [comment]
I will change issue name as well.

**Comment by kamildebicki** [comment]
:+1:

File: .rspec [hunk 1]
```diff
@@ -1,2 +1,4 @@
 --color
 --require spec_helper
+--format documentation
+--order random
```
Comment (student-tomasz) [inline .rspec]: Order is already set to random in `spec/spec_helper.rb` config.

File: spec/models/item_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
+
+RSpec.describe Item, type: :model do
```
Comment (student-tomasz) [inline spec/models/item_spec.rb]: You don't have to specify the type. It's already inferred with config in `spec/rails_helper.rb`.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,75 +1,15 @@
...
-  # compatibility in RSpec 3). It causes shared context metadata to be
-  # inherited by the metadata hash of host groups and examples, rather than
-  # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
-
-  # This allows you to limit a spec run to individual examples or groups
-  # you care about by tagging them with `:focus` metadata. When nothing
-  # is tagged with `:focus`, all examples get run. RSpec also provides
-  # aliases for `it`, `describe`, and `context` that include `:focus`
-  # metadata: `fit`, `fdescribe` and `fcontext`, respectively.
```
Comment (student-tomasz) [inline spec/spec_helper.rb:54]: Why are we removing all those comments? IMO, they're documentation. Would you know how to make use of this feature, without this comment? Just asking for a reason, not that I certainly want to keep these comments.

Comment (agalkiewicz) [inline spec/spec_helper.rb:54]: Maybe you're right and we will need those information. I'll keep it.

Comment (agalkiewicz) [inline spec/spec_helper.rb:54]: But I won't keep comments in rails_helper.rb because I don't find it useful.

File: .rspec [hunk 4]
```diff
@@ -1,2 +1,3 @@
 --color
 --require spec_helper
+--format documentation
```
Comment (gotar) [inline .rspec]: whole group agreed to change it to documentation (you can drop those file from git repo) stop tracking it if every member of team wanna to have own style (config)

File: db/migrate/20160720214821_add_user_ref_to_wishlists.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToWishlists < ActiveRecord::Migration[5.0]
+  def change
+    add_reference :wishlists, :user, foreign_key: true
```
Comment (gotar) [inline db/migrate/20160720214821_add_user_ref_to_wishlists.rb:3]: good to add index for foreign key

File: db/migrate/20160721080417_add_wishlist_ref_to_items.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddWishlistRefToItems < ActiveRecord::Migration[5.0]
+  def change
+    add_reference :items, :wishlist, foreign_key: true
```
Comment (gotar) [inline db/migrate/20160721080417_add_wishlist_ref_to_items.rb:3]: good to add index for foreign key

File: db/schema.rb [hunk 7]
```diff
@@ -21,5 +23,10 @@
     t.text     "description", null: false
     t.datetime "created_at",  null: false
     t.datetime "updated_at",  null: false
+    t.integer  "user_id"
+    t.index ["user_id"], name: "index_wishlists_on_user_id", using: :btree
```
Comment (gotar) [inline db/schema.rb:27]: ok looks like rails add index automatically ;]

**Comment by gotar** [comment]
looks ok, you can merge

**Comment by gotar** [comment]
just squeeze it

**Pull request merged**